apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2.Tests/ReadRowsAcceptanceTest.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Copyright 2017 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Xunit;
+using static Google.Cloud.Bigtable.V2.ReadRowsResponse.Types;
+
+namespace Google.Cloud.Bigtable.V2.Tests
+{
+    public class ReadRowsAcceptanceTest
+    {
+        public static TheoryData<ChunkTestCase> AcceptanceTestCases { get; } = GetAcceptanceTestCases();
+
+        private static TheoryData<ChunkTestCase> GetAcceptanceTestCases()
+        {
+            var typeInfo = typeof(ReadRowsAcceptanceTest).GetTypeInfo();
+            string resourceName = $"{typeInfo.Namespace}.read-rows-acceptance-test.json";
+            using (var reader = new StreamReader(typeInfo.Assembly.GetManifestResourceStream(resourceName)))
+            {
+                var acceptanceTest = JsonConvert.DeserializeObject<AcceptanceTest>(reader.ReadToEnd());
+                var result = new TheoryData<ChunkTestCase>();
+                foreach (var testCase in acceptanceTest.Tests)
+                {
+                    result.Add(testCase);
+                }
+                return result;
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(AcceptanceTestCases))]
+        public async Task Test(ChunkTestCase testCase)
+        {
+            var stream = new MockReadRowsStream(
+                new ReadRowsResponse
+                {
+                    Chunks = { testCase.Chunks.Select(json => CellChunk.Parser.ParseJson(json)) }
+                });
+
+            var responses = new List<Row>();
+            try
```

Comment:
I'd rather see this written more explicitly. I.e. `if (testCase.ExpectsError) Assert.Throws...
To my mind this makes the intent clearer, and makes use of the more suitable Assert statement.

Reply:
Done